### PR TITLE
@/codeforafrica client side search

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

This PR implements client side search using [Fuse.js](https://fusejs.io/). Search is available on the following pages:

 - [x] **Our work**,
 - [x] **Stories**,
 - [x] **About**, and
 - [x] **Opportunities**

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Peek 2022-07-21 15-46

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: apps/codeforafrica/src/components/Articles/Articles.js [hunk 1]
```diff
@@ -1,64 +1,87 @@
...
+
+  const handleChangePage = (_, value) => {
     setPage(value);
   };
-  const [pagination, setPagination] = useState(() => {
-    return computePagination(getFilteredArticles(), page, pageSize);
-  });
+
+  const handleChangeQ = (_, value) => {
+    setQ(value || undefined);
```
Comment (esirK) [inline apps/codeforafrica/src/components/Articles/Articles.js:37]: do we need the `|| undefined`? won't `value` be undefined if not supplied?

**Review by esirK — APPROVED** [review]
LGTM

**Review by kelvinkipruto — COMMENTED** [review]
What if rather than filtering all item based on the search text, display the title and maybe context of the searched text under the search bar, i.e display a few words next or after the text searched.

**Review by kelvinkipruto — APPROVED** [review]
LGTM

File: apps/codeforafrica/src/components/Articles/Articles.js (hunk 1 above)
Comment (kilemensi) [reply apps/codeforafrica/src/components/Articles/Articles.js:37]: Coming from a text field, we get empty string (`""`) instead of `undefined`.

**Comment by kilemensi** [comment]
> What if rather than filtering all item based on the search text, display the title and maybe context of the searched text under the search bar, i.e display a few words next or after the text searched.

Good suggestions @kelvinkipruto but for MVP we try to stick close to the design so that we can give comprehensive feedback and make any/necessary improvements afterwards.

**Pull request merged**